### ISSUE TITLE
Error defining measure column using existing measures

### ISSUE THREAD

**Issue by mikeAdamss** [GSS-Cogs/csvcubed#508]
the measures flavour of this bug: https://app.zenhub.com/workspaces/features-squad-61b72275ac896c0010a1b00b/issues/gss-cogs/csvcubed/490

which was resolved in : https://github.com/GSS-Cogs/csvcubed/commit/5155761d0e399034341e95227e0defa4c8a65195

Some investigation required but it _should_ be as simple as transposing the solution.

- [x] Fix bug via already established approach.
- [x] Add a unit test.
- [x] Check code base for other instances of this issue that haven't bubbled up yet and proactively fix.

**Comment by mikeAdamss** [comment]
as far as I can tell the units and measures are the only places we're using the New Vs Existing component validation approach, the other components are relying on csv-lint and/or csv validation rather than csvcube's own validators.

**Comment by mikeAdamss** [comment]
bugfix pr: https://github.com/GSS-Cogs/csvcubed/pull/509

**Comment by mikeAdamss** [comment]
@robons - I think I've got more bugs inside my bugs, its getting _very_ hard to follow. Can you sanity check the above makes sense to you please, before I go further down this rabbit hole.

**Comment by robons** [comment]
> Fixing this bug invalidates this test:
> 
> https://github.com/GSS-Cogs/csvcubed/blob/dc06019d6e2cfea529a89c9d791a3a5abcc30db1/csvcubed/tests/unit/readers/cubeconfig/v1_0/test_build_friendly_error_msgs.py#L353
> which is currently confirming the incorrect behaviour (it's confirming that a case insensitive measure definition will result in an error, which is the behaviour we're removing with this bug fix), i. e the `NUM` vs `num` declaration issue we've fixed before, but for measures this time.
> 

Yes, unfortunately this does confirm the incorrect behaviour.

> I'm aiming to rewrite this test to confirm the _correct_ behaviour, so failing to declare a new measure that exists within the data will result in a validation error, but its uncovered further bugs, as per the below:
> 
> **Expected behaviour**
> 
> With the following measures config:
> 
> ```
> "Measure": {
>       "type": "measures",
>       "values": [
>         {
>           "label": "Billions"
>         }
>       ]
>     },
> ```
> 
> with this data
> 
> ```
> Measure,Dim-0,Attr-1,Amount,Units
> Billions,1,goods,100,Pounds
> Billions,2,services,321,Pounds
> Billions,3,services,88,Pounds
> Quintillions,4,services,99,Dollars
> ```
> 
> should result in a validation error as Quintillions is a new measure that is undefined.
> 
> **Actual Error**
> 
> Validation Error: The 'Measure' column definition is missing a 'cell_uri_template'; a suitable value could not be inferred.
> 

Really? That's quite surprising. It really should not be identifying the above as a column containing existing measures. Can we have a quick call some time today to confirm this behaviour?

> **Bug Number 1**
> 
> fragile conditional:
> 
> https://github.com/GSS-Cogs/csvcubed/blob/dc06019d6e2cfea529a89c9d791a3a5abcc30db1/csvcubed/csvcubed/utils/qb/validation/observations.py#L160
> 
> We're stating the measures are ExistingQbMeasures if all **the measures that we have** are isinstance(ExistingQbMeasures). This relies on QbMultiMeasureDimension being populated, i.e its failing and misidentifying the measures as Existing as in this instance we've got:
> 
> ```
> QbMultiMeasureDimension(measures=[])
> ```
> 
> i.e you can always confirm that every instance of no instances is ExistingQbMeasures, it would also work it you confirmed all the non existing things were ... anything.
> 
> TODO:
> 
>     * [ ]  Add appropriate code so we never blindly try and confirm all instances of no instances.
> 
>     * [ ]  Check the codebase for other occurrences of fragile all conditionals.
> 
> 
> **Bug Number 2**
> 
> We shouldn't be allowing the creation of QbMultiMeasureDimension with no measures in it.
> 
> TODO:
> 
>     * [ ]  How and where it's happening in this instance, fix immediate problem.
> 
>     * [ ]  Update csvcubed validation to let us know when this does happen, create appropriate test coverage.

I think if you enforce that `QbMultiMeasureDimension` and `QbMultiUnits` must have at least one measure/unit defined, then it'd mean that bug number 1 can never happen. That's the fix I'd suggest doing.
